tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_notification.py

```diff
@@ -18,12 +18,19 @@ def test_sms_notification(self):
 
         system = System(self.marionette)
 
+        self.assertEqual(0, len(self.data_layer.get_all_sms()))
+
         # Send a SMS to the device
         self.data_layer.send_sms(self.environment.phone_numbers[0], _text_message_content)
 
-        # We will wait upto 300 seconds for the SMS to arrive due to network latency
-        system.wait_for_notification_toaster_displayed(timeout=300,
-                    message="Notification did not appear. SMS database dump: %s " % self.data_layer.get_all_sms())
-        system.wait_for_notification_toaster_not_displayed()
+        # self.data_layer.send_sms can be so slow that the sms has already been received,
+        # That's why the following if statement, see also bug 1172460
+        if len(self.data_layer.get_all_sms()) == 1:
```

Comment:
The problem is that self.data_layer.send_sms is too slow, it seems. When that has finished sometimes, the notification might have already appeared and disappeared. It makes me wonder why we don't use the normal UI here for sending an sms.